### PR TITLE
Provide modern example enctypes in ipa-getkeytab(1)

### PR DESCRIPTION
**Pull request by frozencemetery**

### CODE REVIEW

File: client/man/ipa-getkeytab.1 [hunk 1]
```diff
@@ -124,10 +116,10 @@ against a FreeIPA server more recent than version 3.3. The user requesting the
 keytab must have access to the keys for this operation to succeed.
 .SH "EXAMPLES"
 Add and retrieve a keytab for the NFS service principal on
-the host foo.example.com and save it in the file /tmp/nfs.keytab and retrieve just the des\-cbc\-crc key.
+the host foo.example.com and save it in the file /tmp/nfs.keytab and retrieve just the aes256\-sha2 key.
 
 .nf
-   # ipa\-getkeytab \-p nfs/foo.example.com \-k /tmp/nfs.keytab \-e des\-cbc\-crc
+   # ipa\-getkeytab \-p nfs/foo.example.com \-k /tmp/nfs.keytab \-e aes\-sha2
```
Comment (rcritten) [inline client/man/ipa-getkeytab.1]: aes256-sha2 is not a valid key name.

IPA supports:

krbSupportedEncSaltTypes: aes256-cts:normal
krbSupportedEncSaltTypes: aes256-cts:special
krbSupportedEncSaltTypes: aes128-cts:normal
krbSupportedEncSaltTypes: aes128-cts:special
krbSupportedEncSaltTypes: camellia128-cts-cmac:normal
krbSupportedEncSaltTypes: camellia128-cts-cmac:special
krbSupportedEncSaltTypes: camellia256-cts-cmac:normal
krbSupportedEncSaltTypes: camellia256-cts-cmac:special

Comment (frozencemetery) [reply client/man/ipa-getkeytab.1]: Why doesn't IPA support the SHA2 enctypes?  Is this tracked somewhere?

Comment (abbra) [reply client/man/ipa-getkeytab.1]: This is not IPA problem -- `ipa-getkeytab` reports default encryption types from MIT Kerberos library.

```
# ipa-getkeytab --permitted-enctypes
Supported encryption types:
AES-256 CTS mode with 96-bit SHA-1 HMAC
AES-256 CTS mode with 192-bit SHA-384 HMAC
Camellia-256 CTS mode with CMAC
AES-128 CTS mode with 96-bit SHA-1 HMAC
AES-128 CTS mode with 128-bit SHA-256 HMAC
Camellia-128 CTS mode with CMAC
```
This comes from the following snippet in `client/ipa-getkeytab.c`:
```
        if (ret == -1 && permitted_enctypes &&
            !(server || principal || keytab || quiet)) {
                krb5_enctype *ktypes;
                char enc[79]; /* fit std terminal or truncate */

                krberr = krb5_get_permitted_enctypes(krbctx, &ktypes);
                if (krberr) {
                        fprintf(stderr, _("No system preferred enctypes ?!\n"));
                        exit(1);
                }
                fprintf(stdout, _("Supported encryption types:\n"));
                for (i = 0; ktypes[i]; i++) {
                        krberr = krb5_enctype_to_string(ktypes[i], enc, 79);
                        if (krberr) {
                                fprintf(stderr, _("Warning: "
                                        "failed to convert type (#%d)\n"), i);
                                continue;
                        }
                        fprintf(stdout, "%s\n", enc);
                }
                ipa_krb5_free_ktypes(krbctx, ktypes);
                exit (0);
        }
```

**Comment by abbra** [comment]
Also, `fedora-30/test_advise` failed due to TLSv1 alert in communicating to Dogtag backend.

File: client/man/ipa-getkeytab.1 (hunk 1 above)
Comment (rcritten) [reply client/man/ipa-getkeytab.1]: The permitted encrypts are hardcoded in IPA install via install/share/kerberos.ldif

Comment (frozencemetery) [reply client/man/ipa-getkeytab.1]: Trying to understand... are the sha1 and sha2 enctypes being grouped together for some reason?  What's the right way to request (using krb5 enctype names) a keytab with aes256-cts-hmac-sha384-192 key?  with aes128-cts-hmac-sha256-128?

Comment (rcritten) [reply client/man/ipa-getkeytab.1]: Like I said, they aren't enabled currently. The enctypes need to be added in LDAP. I'll open a separate PR on that.

Comment (rcritten) [reply client/man/ipa-getkeytab.1]: https://github.com/freeipa/freeipa/pull/3841

Comment (frozencemetery) [reply client/man/ipa-getkeytab.1]: Sorry @rcritten - missed your comment and was replying to @abbra.  Thanks for opening the PR!

**Comment by abbra** [comment]
LGTM

**Comment by abbra** [comment]
I think this PR should be pushed now that PR#3841 is pushed.

**Comment by rcritten** [comment]
camellia should probably be added to the man page as well while changes are being made.

Should probably hold off on this one until the backport of 3841 gets merged.

**Comment by frozencemetery** [comment]
Added aes-sha2 and camellia types, then rebased onto master.

File: client/man/ipa-getkeytab.1 [hunk 2]
```diff
@@ -88,11 +88,7 @@ This options returns a description of the permitted encryption types, like this:
 Supported encryption types:
 AES\-256 CTS mode with 96\-bit SHA\-1 HMAC
 AES\-128 CTS mode with 96\-bit SHA\-1 HMAC
-Triple DES cbc mode with HMAC/sha1
```
Comment (rcritten) [inline client/man/ipa-getkeytab.1]: Add AES 128/256 SHA-2 here as well?

AES-128 CTS mode with 128-bit SHA-256 HMAC
AES-256 CTS mode with 192-bit SHA-384 HMAC

Comment (frozencemetery) [reply client/man/ipa-getkeytab.1]: Ah, yes, thank you.

**Comment by rcritten** [comment]
master:

* 3cb9444c4c9af126c0559d4f1cb6c58aece78b19 Provide modern example enctypes in ipa-getkeytab(1)

**Pull request closed without merging**